example_concat_zipcsv.py

```diff
@@ -2,11 +2,40 @@
 import pandas as pd
 import zipfile
 
-dfs = []
-with zipfile.ZipFile('F:/Downloads/Compressed/Rulo_5SW.zip') as z:     #   z.infolist()
-    for d in pd.date_range('2019-08-24', '2019-08-30', freq='D'):
-        with z.open('Rulo_5SW/Rulo_5SW_soil_{:%Y%m%d}_0703.csv'.format(d)) as f:
-            dfs.append(pd.read_csv(f, header=0, skiprows=1).iloc[2:])
-        
-
-df_all = pd.concat(dfs).reset_index(drop=True)
+def openByDate(zipfpath, statedate, enddate):
+    '''
+    open zip and read files within the range of the starting date and ending date
+    preferable, controllable
+    '''
+    dfs = []
+    with zipfile.ZipFile(zipfpath) as z:     #   z.infolist()
+        filenames = [f.filename f in z.infolist()]
+        for d in pd.date_range(startdate, enddate, freq='D'):
+            toOpen = 'Rulo_5SW/Rulo_5SW_soil_{:%Y%m%d}_0703.csv'.format(d)
+            if toOpen in filenames:
+                # check if the file exists in the zip
+                with z.open(toOpen) as f:
+                    dfs.append(pd.read_csv(f, header=0, skiprows=1).iloc[2:])
+    return pd.concat(dfs).reset_index(drop=True)
+
+
+def openAllExisting(zipfpath):
+    '''
+    passively open all files. may open some files you don't want and make cleaning up more difficult
+    '''
+    
+    soil = []
+    agl = []
+    with zipfile.ZipFile(zipfpath) as z:     #   
+        for f in z.infolist():
+            fname = f.filename
+            if fname.endswith('csv'):
+                # check if it is a CSV file then open it
+                with z.open(fname) as f:
+                    df = pd.read_csv(f, header=0, skiprows=1).iloc[2:]
+                if 'soil' in fname:
+                    soil.append(df)
+                if 'agl' in fname:
+                    agl.append(df)
+
+    return pd.concat(agl).reset_index(drop=True), pd.concat(soil).reset_index(drop=True)
```

Comment:
I run the function as:

`openAllExisting(zipfpath)`

How do I then reference each separate df? soil and agl? 

If I call

`print(df)`

it appears I get only the soil data.
How can I call them separately?

Reply:
Also, when I run the function code, I get the following error:
`C:\Users\lthompson8\Miniconda3\lib\site-packages\IPython\core\interactiveshell.py:3254: DtypeWarning: Columns (1,2,3,5,7,8,10,12,13,15,17,18,20,21,23,24,25,26,27,29,30,31,33,35,36,37,38,40,42,43,44) have mixed types.Specify dtype option on import or set low_memory=False.
  if (await self.run_code(code, result,  async_=asy)):
Out[331]: `

This is the same error I was getting with the code I originally had as follows:
`extension = 'csv'
all_filenames = [i for i in glob.glob('*.{}'.format(extension))]
#combine all files in the list
combined_csv = pd.concat([pd.read_csv(f)for f in all_filenames ])
#export to csv
combined_csv.to_csv( "combined_soil_test1.csv", index = False, encoding='utf-8-sig')`